webapp/src/immersivereader.tsx

```diff
@@ -46,29 +60,98 @@ function beautifyText(content: string): string {
         );
     }
 
-    // Remove ** around bold text, replace with "
-    function cleanBold(content: string): string {
+    // If the ```block ``` annotation is used to display a block in
+    // a modal, remove it
+    function cleanHintText(content: string): string {
         return content.replace(
-            /\*\*([^*]+)\*\*/gu,
-            (matched, word, offset, s) => lf("\"{0}\"", word)
+            /^```block$[^`]*^```$/gum,
+            ""
+        );
+    }
+
+    // Remove ** and __ around bold text, replace with " if it's only one word
+    function cleanBold(content: string): string {
+
+        let singleWordBold = content.replace(
+            /\*\*([^*^\s]+)\*\*|__([^\s^_]+)__/gu,
+            (matched, boldStar, boldUnder, offset, s) => {
+                return "\"" + lf("{0}", (boldStar ? boldStar : boldUnder)) + "\""
+            }
+        )
+
+        return singleWordBold.replace(
+            /\*\*([^*]+)\*\*|__([^_]+)__/gu,
```

Comment:
not blocking for this PR, but this doesn't catch something like https://github.com/microsoft/pxt-arcade/blame/master/docs/skillmap/platformer/activity2.md#L209 which has italicized text inside a bold. we could also just modify the tutorial to work, though!

Reply:
Ah yeah :\ I couldn't think of a good way to catch this case, and hadn't seen many places where it was used >> I can put up a PR to modify the tutorial

Reply:
not urgent, i think!